app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,38 @@
 class PostsController < ApplicationController
   def index
+    @current_user = current_user
     @posts_per_page = 2
     @user = User.find(params[:user_id])
     @page = params.fetch(:page, 1)
     @posts = @user.posts[2 * (@page.to_i - 1), @posts_per_page]
   end
 
   def show
+    @current_user = current_user
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @current_user = current_user
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    post = Post.new(params.require(:user_posts).permit(:title, :text))
+    post.author = current_user
+    respond_to do |format|
+      format.html do
+        if post.save
+          flash[:success] = 'Post was successfully created'
+          redirect_to user_path(current_user)
+        else
+          flash.now[:error] = 'Error: Post could not be saved'
+          render :new, new_user_post_path(current_user)
+        end
```

Comment:
- [ ] please have a look at [this article](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) for an example of how to handle params in the controller.

Reply:
I can’t understand what’s wrong with the way I’m using the parameter. Please, could you be more specific?